lttnganalyses/cli/mi.py

```diff
@@ -281,7 +242,48 @@ def _eq(self, other):
         return self._value == other._value
 
 
-class _SimpleName(_StructuredDataObject):
+class Boolean(_SimpleValue):
+    CLASS = 'bool'
+
+
+NEG_INF = '-inf'
+POS_INF = '+inf'
+
+
+class Number(_SimpleValue):
+    CLASS = 'number'
+
+    def __init__(self, value, low=None, high=None):
+        super().__init__(value)
+        self._low = low
+        self._high = high
+
+    def _to_native_object(self):
+        obj = {}
+
+        if self.value is not None:
+            obj['value'] = self.value
+
+        if self._low is not None:
+            obj['low'] = self._low
+
+        if self._high is not None:
+            obj['high'] = self._high
+
+        return obj
+
+    def _eq(self, other):
+        self_tuple = (self.value, self._low, self._high)
+        other_tuple = (other.value, other._low, other._high)
```

Comment:
I'm not quite sure of how we want to define equality here. I feel like simple comparing `value` might also be valid, depending on the case.